### ISSUE TITLE
worker.close hangs when using limiter

### ISSUE THREAD

**Issue by sakari** [taskforcesh/bullmq-pro-support#56]
with bullmq-pro-6.3.2 worker.close hangs (or at least takes a long time) when using rate limiting and not forcing

```
      const queueName = 'some queue';
      const queue = new QueuePro(queueName, { connection: redisConnectionConfiguration });
      const limiter = { max: 1, duration: 60 * 1000 };
      const worker = new WorkerPro(queueName, async () => {}, { connection: redisConnectionConfiguration, limiter });
      await queue.add(anyString(), 1);
      await promisify(setTimeout)(1000);
      await worker.close();
``` 

with worker.close(true) we do not hang but I would like to allow the currently processed job to finish.

**Comment by manast** [comment]
But is it that the Worker is still working on some job and that's why it takes so long time?

**Comment by sakari** [comment]
> async () => {}

is the worker body. That *should* not take many moons to go through 😁

**Comment by manast** [comment]
ah sorry didn't see the code 😂

**Comment by manast** [comment]
I have been able to reproduce it and it seems to be present in BullMQ as well. Debugging it now...

**Comment by manast** [comment]
The issue should be fixed now in Pro version v6.3.4.
